presto-main/src/main/java/com/facebook/presto/util/LongValueCompressor.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.util;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class LongValueCompressor
```

Comment:
I think this is overkill. The functionality we need is specific to the operator that assigns ids, so I don't see much value in abstracting it at this point. Also, this is very performance-sensitive code, so we should be careful how it's structured.

All you need is something like this:
```java
long makeId(int stageId, int taskId, int rowId)
{
    checkArgument(stageId < xxx);
    checkArgument(taskId < yyy);
    checkArgument(rowId < zzz);

    return (((long) stagedId) << aaa) | (((long) taskId) << bbb) | (long) rowId;
}
```

Given that stageId and taskId are constant for an instantiation of the operator, you could even do:

```java
long makeId(long prefix, long rowId) 
{
    checkArgument(rowId < zzz);
    return prefix | rowId;
}
```

and pre-calculate the prefix in the constructor of the operator.